numpy/lib/histograms.py

```diff
@@ -345,6 +345,97 @@ def _search_sorted_inclusive(a, v):
         a.searchsorted(v[-1:], 'right')
     ))
 
+def histogram_bin_edges(a, bins=10, range=None, weights=None):
+    """
+    Function to calculate only the edges of the bins used by the histogram function. 
+
+    Parameters
+    ----------
+    a : array_like
+        Input data. The histogram is computed over the flattened array.
+    bins : int or sequence of scalars or str, optional
+        If `bins` is an int, it defines the number of equal-width
+        bins in the given range (10, by default). If `bins` is a
+        sequence, it defines the bin edges, including the rightmost
+        edge, allowing for non-uniform bin widths.
+
+        .. versionadded:: 1.11.0
+
+        If `bins` is a string from the list below, `histogram` will use
+        the method chosen to calculate the optimal bin width and
+        consequently the number of bins (see `Notes` for more detail on
+        the estimators) from the data that falls within the requested
+        range. While the bin width will be optimal for the actual data
+        in the range, the number of bins will be computed to fill the
+        entire range, including the empty portions. For visualisation,
+        using the 'auto' option is suggested. Weighted data is not
+        supported for automated bin size selection.
+
+        'auto'
+            Maximum of the 'sturges' and 'fd' estimators. Provides good
+            all around performance.
+
+        'fd' (Freedman Diaconis Estimator)
+            Robust (resilient to outliers) estimator that takes into
+            account data variability and data size.
+
+        'doane'
+            An improved version of Sturges' estimator that works better
+            with non-normal datasets.
+
+        'scott'
+            Less robust estimator that that takes into account data
+            variability and data size.
+
+        'rice'
+            Estimator does not take variability into account, only data
+            size. Commonly overestimates number of bins required.
+
+        'sturges'
+            R's default method, only accounts for data size. Only
+            optimal for gaussian data and underestimates number of bins
+            for large non-gaussian datasets.
+
+        'sqrt'
+            Square root (of data size) estimator, used by Excel and
+            other programs for its speed and simplicity.
```

Comment:
I wonder whether we should remove this help from `np.histogram`, and leave it here exclusively?

Reply:
It is a bit redundant to have the different types of bins documented twice. If you'd like I'll remove it form ```np.histogram``` and then push the commit with all the fixes.

Reply:
Let's see if anyone else has anything to say about that first

Reply:
Sure thing. I'll push the latest commit leaving the doc for ```np.histogram``` intact. In case you want it changed later, I'll modify the docstring.

Reply:
I face a somewhat similar dilemma with `np.quantile` and `np.percentile` (and `np.nanquantile`,  `np.nanpercentile`) in #9211.  I think in #9213 there's a comment about `just make the usual docstring`, so I went the verbose way.

Reply:
Let's leave the duplication, and then we can propose removing it in a later PR

Reply:
@eric-wieser I think it is a good idea to remove the duplication. Would it make sense to have a separate documentation page for histogram bin estimation? This function and histogram could link to it.

Reply:
Already removed in #10755